### ISSUE TITLE
Firefox Extension Support

### ISSUE THREAD

**Issue by Synthawk** [FranciscoAT/saltyboy#102]
This is more a feature request than a bug. Any chance for Firefox extension support? I know I'm not the only one actively trying to move away from Chrome and Chromium based browsers. Would be nice to have this available, I find it much better than Saltbot.

**Comment by darklunch** [comment]
Hi @how2vim and @Synthawk,

I am a passionate Firefox user and love the SaltyBoy project, so I wanted to help out by taking a stab at the Firefox port.

I have a fork working successfully on Firefox (Manifest V3), but getting there required a few architectural changes that I wanted to share context on.

While Chrome is currently lenient with cross-origin requests in Manifest V3 content scripts, Firefox is much stricter. When I first ported the manifest, the extension would load but fail to retrieve match data (returning undefined) because the content script was being blocked by CORS policies on saltybet.com.

To fix this and ensure stability across both browsers, I had to refactor the data fetching logic:

    Manifest Updates: Added the required browser_specific_settings (Gecko ID) and explicit host_permissions for the SaltyBoy API.

    Background Proxy: I moved the actual fetch calls from the content script to the background script. The content script now sends a message to the background service, which executes the request (since it has the privileged host permissions) and returns the data. This bypasses the CORS restrictions on the content script.

I've tested this on a fresh install and it is placing bets and displaying overlay data perfectly in Firefox.

You can check out the changes on my fork here: [https://github.com/darklunch/saltyboy-ff](https://github.com/darklunch/saltyboy-ff)

@how2vim If this approach sounds good to you, I'd be happy to open a Pull Request. Since I'll be using this daily, I am also more than happy to maintain the Firefox compatibility layer and handle any Firefox-specific issues or releases going forward so you don't have to worry about the extra overhead.

(P.S. I might also fork the backend components and try to do some other statistical voodoo down the line!)
